### DISCUSSION TITLE
Just for curiosity

### DISCUSSION THREAD

**Discussion by ozw1z5rd** [linkedin/shiv#180] [General]
py34 is not supported, are there some blocking issues in porting it?  
If a masochist wants to port this to python 3.4, will he suffer and succeed  or will unable to complete the work?

**Comment by lorencarvalho** [comment]
Hi @ozw1z5rd,

_Theoretically_ you could create a `pyz` that runs fine on Py3.4, but you'd have to use 3.6+ to create it. This is because `shiv` itself uses the PEP 526 variable type hint syntax that is only supported in 3.6+ (https://docs.python.org/3/whatsnew/3.6.html#whatsnew36-pep526), whereas the bootstrap code that is injected into the resulting binary is technically compatible with any 3.x release (I believe, I haven't tested) as it does not contain any inline type hints. Easy enough to do, though:

```
python3.6 -m shiv -o ~/out.pyz httpx -p "/usr/bin/env python3.4"
```

> If a masochist wants to port this to python 3.4, will he suffer and succeed or will unable to complete the work?

They may suffer, they may succeed, but I probably wouldn't merge it because I like the PEP 526 syntax. I've even considered adding type hints to the bootstrap module, and may still at some point.

**Reply by ozw1z5rd** [reply]
Thanks for the quick answer, I'll give a try.
